learning/tour-of-beam/frontend/lib/pages/welcome/screen.dart

```diff
@@ -68,21 +72,25 @@ class _WideWelcome extends StatelessWidget {
 }
 
 class _NarrowWelcome extends StatelessWidget {
-  const _NarrowWelcome();
+  final WelcomeNotifier notifier;
+
+  const _NarrowWelcome(this.notifier);
 
   @override
   Widget build(BuildContext context) {
     return Column(
-      children: const [
-        _SdkSelection(),
-        _TourSummary(),
+      children: [
+        _SdkSelection(notifier),
+        _TourSummary(notifier),
       ],
     );
   }
 }
 
 class _SdkSelection extends StatelessWidget {
-  const _SdkSelection();
+  final WelcomeNotifier notifier;
```

Comment:
Is it possible to create an analog of BlocBuilder so that there is no need to pass around notifiers to get state?

Reply:
Only direct passing provides a compile-time safety.
Anything like Provider etc. without us knowing before runtime.
We should prefer direct passing unless there is a good reason not to in a particular case.

In our projects, I see only theme as a good usecase, not event `PlaygroundController`.